HARK/ConsumptionSaving/ConsIndShockModel.py

```diff
@@ -2341,6 +2346,301 @@ def calc_bounding_values(self):
+        if self.cycles == 0: 
+            
+            if shk_dstn == None:
+                shk_dstn = self.IncShkDstn
+            
+            dist_mGrid = self.dist_mGrid #Grid of market resources
+            dist_pGrid = self.dist_pGrid #Grid of permanent incomes
+            aNext = dist_mGrid - self.solution[0].cFunc(dist_mGrid)  #assets next period
+            
+            self.aPol_Grid = aNext # Steady State Asset Policy Grid
+            self.cPol_Grid = self.solution[0].cFunc(dist_mGrid) #Steady State Consumption Policy Grid
+            
+            # Obtain shock values and shock probabilities from income distribution
+            bNext = self.Rfree*aNext # Bank Balances next period (Interest rate * assets)
+            shk_prbs = shk_dstn[0].pmf  # Probability of shocks 
+            tran_shks = shk_dstn[0].X[1] # Transitory shocks
+            perm_shks = shk_dstn[0].X[0] # Permanent shocks
+            LivPrb = self.LivPrb[0] # Update probability of staying alive
+            
+            #New borns have this distribution (assumes start with no assets and permanent income=1)
+            NewBornDist = jump_to_grid(tran_shks,np.ones_like(tran_shks),shk_prbs,dist_mGrid,dist_pGrid)
+            
+            
+            if len(dist_pGrid) == 1: 
+                
+                NewBornDist = jump_to_grid_fast(tran_shks,shk_prbs,dist_mGrid)
+
+                # Generate Transition Matrix
+                TranMatrix = np.zeros((len(dist_mGrid),len(dist_mGrid)))
+                for i in range(len(dist_mGrid)):
+                        mNext_ij = bNext[i]/perm_shks + tran_shks # Compute next period's market resources given todays bank balances bnext[i]
+                        TranMatrix[:,i] = LivPrb*jump_to_grid_fast(mNext_ij, shk_prbs,dist_mGrid) + (1.0-LivPrb)*NewBornDist
+                self.tran_matrix = TranMatrix
+                
+            else:
+                NewBornDist = jump_to_grid(tran_shks,np.ones_like(tran_shks),shk_prbs,dist_mGrid,dist_pGrid)
+                # Generate Transition Matrix
+                TranMatrix = np.zeros((len(dist_mGrid)*len(dist_pGrid),len(dist_mGrid)*len(dist_pGrid)))
+                
+                for i in range(len(dist_mGrid)):
+                    for j in range(len(dist_pGrid)):
+                        mNext_ij = bNext[i]/perm_shks + tran_shks # Compute next period's market resources given todays bank balances bnext[i]
+                        pNext_ij = dist_pGrid[j]*perm_shks # Computes next period's permanent income level by applying permanent income shock
+                        TranMatrix[:,i*len(dist_pGrid)+j] = LivPrb*jump_to_grid(mNext_ij, pNext_ij, shk_prbs,dist_mGrid,dist_pGrid) + (1.0-LivPrb)*NewBornDist
+    
+                self.tran_matrix = TranMatrix
+            
+            
+        elif self.cycles > 1:
+            raise Exception('calc_transition_matrix requires cycles = 0 or cycles = 1')
+            
+        elif self.T_cycle!= 0 :
```

Comment:
Is there a model that has cycles==1 and T_cycle==0? I'm thinking this should never happen, and if it does, it's probably a bug.

Reply:
I'm not sure, that is why I was so specific on the conditional cases.